### DISCUSSION TITLE
Kong pg_ro_host property

### DISCUSSION THREAD

**Discussion by RagunathInGIT** [Kong/kong#7692] [Help] [answered]
Team,

We are running Kong on K8s with Postgres DB. We are running Kong in two DataCenters with PG master running in one DC and its replica running in another DC. Both DC's points to same master instance for read-write. In order to avoid latency, we configured pg_ro_host property in the secondary Datacenter so that read calls will go to local replica instance.

Problem we see here is, if at all the replica instance goes down for some reasons, though the Kong pods remains up, all the read-write calls are all failing as well as the read calls where we have configured one of the security plugins like OAuth, BasicAuth etc. If the API is  not secured with any security plugin such as HealthCheck API, that works fine. So it looks like Kong is not falling back to pg_host property for all  read, write calls if pg_ro_host property is un-reachable. 

Could you please clarify if this is how Kong behaves? That is, does it require pg_ro_host be available all the time?

**Comment by kikito** [comment, marked as answer]
Hello, I confirm that is how `pg_ro_host` works at the moment. If it's configured, the database must respond when queried, otherwise an error will be raised. There is no "fallback to r/w db".

However, if the data that is to be read is found in the cache, Kong sometimes will use that value in the cache without even attempting to read from the database.
